### ISSUE TITLE
[data-grid] Support multiple views?

### ISSUE THREAD

**Issue by oliviertassinari** [mui-org/material-ui-x#3520]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

Have the ability to save different views on how the data set is displayed.

### Examples 🌈

On https://app.qonto.com/

<img width="553" alt="Screenshot 2021-12-28 at 17 30 43" src="https://user-images.githubusercontent.com/3165635/147586982-178fd062-ef9a-4dcb-9605-7d1ba77579ea.png">

On https://airtable.com/

<img width="329" alt="Screenshot 2021-12-28 at 17 32 24" src="https://user-images.githubusercontent.com/3165635/147587091-aeb26e37-d473-479d-8cec-e8da820e8c52.png">

On https://zendesk.com/

<img width="311" alt="Screenshot 2021-12-28 at 17 34 54" src="https://user-images.githubusercontent.com/3165635/147587341-700f7200-fad3-4eec-b10b-60b425944248.png">

### Motivation 🔦

There are often multiple ways to look at the same dataset. Depending on the specific use case you have, you might want to a different filter, sort order, column order, column, visibility, etc.

Considering that developers often ask about the ability to persist these params so their users can restore the data grid to the same state between two sessions, I wonder if it would make sense to have built-in support for custom views. I'm personally using this feature extensively on Airtable and Zendesk.

I have left the "waiting for upvotes" label, as I couldn't find any data grid doing it, from [our benchmark list](https://www.notion.so/mui-org/Data-Grid-component-7e0d4fb7a10943c9aac340198684197c). So maybe the idea is a dead end 😁

**Comment by cq-ubaid-khan** [comment]
I raised the same kind of feature request few days ago see #3448.

**Comment by oliviertassinari** [comment]
Blocked by #820 and #3448

@cq-ubaid-khan It's close, but it's different. For the feature I'm proposing here to be relevant, we first need developers to persist the state of the data grid between sessions. Without it would be useless (As I user, which would I create 2 views if tomorrow, I lose them?)

**Comment by cq-ubaid-khan** [comment]
> Ok, so blocked by #820 and #3448. Added:
> 
> <img alt="Screenshot 2021-12-29 at 18 10 10" width="879" src="https://user-images.githubusercontent.com/3165635/147686817-ab85f925-e250-43de-8ec2-8ccaaa022865.png">
> 
> @cq-ubaid-khan It's close, but it's different. For the feature I'm proposing here to be relevant, we first need developers to persist the state of the data grid between sessions. Without it, the feature would be useless (As an end-user, why would I create two views if, in my next session, I lose them?)

If you see #3448, I've mentioned that I want to persist the state in persistent location so later when we revisit the page we get the same settings and currently I've achieve that by creating an object that maps all of the datagrid settings to object and save into database and when revisit the page I can choose from different layouts to get the settings object from database and apply them to datagrid.
with that in place I can create as many views as I want. see [3448#issuecomment-1000143967](https://github.com/mui-org/material-ui-x/issues/3448#issuecomment-1000143967)

![image](https://user-images.githubusercontent.com/58769931/147744381-9a56dbf1-fc93-4e31-9b49-55dce0fa4467.png)

However, I had to write a bunch of states mapping code, It would be nice to have this feature out of the box.

Thanks.
